### PR TITLE
Echo|Print*Fixer - unify printing fixers

### PR DESCRIPTION
**Pull request by SpacePossum**
Early PR open as POC, see how the tests do and to get the discussion going on naming things among the community.

### CODE REVIEW

File: src/Fixer/Alias/NoMixedEchoPrintFixer.php [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        if (null === $configuration) {
+            $configuration = self::$defaultConfig;
+        } else {
+            if (1 !== count($configuration) || !array_key_exists('use', $configuration) || !in_array($configuration['use'], array('print', 'echo'), true)) {
+                throw new InvalidFixerConfigurationException($this->getName(), sprintf(
+                    'Expected array of element "use" with value "echo" or "print", got "%s".',
```
Comment (SpacePossum) [inline src/Fixer/Alias/NoMixedEchoPrintFixer.php]: missing utest for exception thrown

File: src/Fixer/Alias/NoMixedEchoPrintFixer.php [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+     */
+    private $candidateTokenType;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        if (null === $configuration) {
+            $configuration = self::$defaultConfig;
```
Comment (SpacePossum) [inline src/Fixer/Alias/NoMixedEchoPrintFixer.php]: missing utest for using default config

File: tests/Fixer/Alias/NoMixedEchoPrintFixerTest.php [hunk 3]
```diff
@@ -16,23 +16,138 @@
...
+                print ("test");
+                ',
+            ),
+            array(
+                '<?php
+                print("test");
+                ',
+            ),
+            // echo can take multiple parameters (although such usage is rare) while print can take one argument
+            // http://www.w3schools.com/php/php_echo_print.asp
```
Comment (SpacePossum) [inline tests/Fixer/Alias/NoMixedEchoPrintFixerTest.php]: note to self to update the link from w3schools to a formal source

**Comment by keradus** [comment]
:+1:, should be green now

File: tests/Fixer/Alias/NoMixedEchoPrintFixerTest.php [hunk 4]
```diff
@@ -282,8 +285,11 @@ public function testDefaultConfig()
      */
     public function testWrongConfig($wrongConfig)
     {
-        $this->expectException('PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException');
-        $this->expectExceptionMessageRegExp('#^\[no_mixed_echo_print\] Expected array of element \"use\" with value \"echo\" or \"print\", got \"array \([\S\s]*\)\".$#');
+        $this->setExpectedExceptionRegExp(
```
Comment (SpacePossum) [inline tests/Fixer/Alias/NoMixedEchoPrintFixerTest.php]: :+1:  (although `setExpectedExceptionRegExp` is deprecated in fav. for `ExpectedExceptionRegExp` IIRC)

Comment (keradus) [inline tests/Fixer/Alias/NoMixedEchoPrintFixerTest.php]: `->setExpectedException*` method do exist in bot phpunit versions, even if they are deprecated.
`->expectException*` methods do not exist on lower phpunit version used by lowers PHP engines, so you cannot use it. as simple as that.

Comment (SpacePossum) [inline tests/Fixer/Alias/NoMixedEchoPrintFixerTest.php]: we could polyfil ^_^  but fair enough, good catch!

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request closed without merging**